tsp-login/src/main/java/pl/edu/agh/iet/tsp/login/oauth2/configuration/facebook/FacebookClientConfiguration.java

```diff
@@ -0,0 +1,44 @@
+package pl.edu.agh.iet.tsp.login.oauth2.configuration.facebook;
+
+import org.springframework.boot.autoconfigure.security.oauth2.resource.ResourceServerProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.oauth2.client.token.grant.code.AuthorizationCodeResourceDetails;
+import pl.edu.agh.iet.tsp.login.oauth2.configuration.ClientConfiguration;
+
+/**
+ * @author Bartłomiej Grochal
+ */
+@Configuration
+public class FacebookClientConfiguration implements ClientConfiguration {
+
+    @Override
+    public AuthorizationCodeResourceDetails getClientProperties() {
+        return getFacebookClientProperties();
+    }
```

Comment:
Właśnie niestety nie, też tak próbowałem pierwotnie i okazało się, że nagle wszystkie możliwe logowania przechodziły tylko przez jednego providera. Wydaje mi się, że w takim podejściu problem jest z tym, że konstruujemy kilka beanów o typie `AuthorizationCodeResourceDetails`, więc ostatni bean nadpisuje poprzednie (bo gdybyśmy mieli kilka nierozróżnialnych beanów w kontenerze, to nie wiedzielibyśmy, który dokładnie pobrać). Gdzieś w sieci widziałem podobny problem i workaround na niego był analogiczny do tego, co napisałem.